### PR TITLE
docs(python): Update `then` and `otherwise` docstrings with "strings are parsed as column names"

### PR DESCRIPTION
**Pull request by henryharbeck**
Ref https://github.com/pola-rs/polars/issues/13611

String inputs being parsed as literals for `when`, `then` and `otherwise` was removed in the last breaking release. This PR updates the docstrings for `then` and `otherwise` to note this. Updated wording is consistent with other existing docstrings.

Also has very minor updates in `when` docstrings with:
- `colname = value` -> `col_name = value`
- `when, thens` -> when-then operations ~`when, then`s~

### CODE REVIEW

**Review by stinodego — APPROVED** [review]
Good catch, thanks!

File: py-polars/polars/functions/whenthen.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def when(
     │ 4   ┆ 0   ┆ 1   │
     └─────┴─────┴─────┘
 
-    Or with multiple `when, thens` chained:
+    Or with multiple `when, then`s chained:
```
Comment (stinodego) [inline py-polars/polars/functions/whenthen.py:69]: I think this syntax breaks Sphinx.

```suggestion
    Or with multiple when-then operations chained:
```

File: py-polars/polars/functions/whenthen.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def when(
     `pl.when(<condition>).then(<value if condition>)`., and optionally followed by
     chaining one or more `.when(<condition>).then(<value>)` statements.
 
-    Chained `when, thens` should be read as Python `if, elif, ... elif` blocks, not as
+    Chained `when, then`s should be read as Python `if, elif, ... elif` blocks, not as
```
Comment (stinodego) [inline py-polars/polars/functions/whenthen.py:27]: Same here.

File: py-polars/polars/functions/whenthen.py (hunk 1 above)
Comment (henryharbeck) [reply py-polars/polars/functions/whenthen.py:69]: I figured that was it once I saw that build break. Sorry about that, will fix up now.

**Pull request merged**